OpenRA.Mods.Common/Traits/Render/WithSiloSpriteBody.cs

```diff
@@ -0,0 +1,76 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits.Render
+{
+	[Desc("Render trait for buildings that change the sprite according to the remaining resource storage capacity across all depots.")]
+	public class WithSiloSpriteBodyInfo : WithSpriteBodyInfo
+	{
+		[Desc("Internal resource stages. Does not have to match number of sequence frames.")]
+		public readonly int Stages = 10;
+
+		public override object Create(ActorInitializer init) { return new WithSiloSpriteBody(init, this); }
+
+		public override IEnumerable<IActorPreview> RenderPreviewSprites(ActorPreviewInitializer init, RenderSpritesInfo rs, string image, int facings, PaletteReference p)
+		{
+			if (!EnabledByDefault)
+				yield break;
+
+			var anim = new Animation(init.World, image);
+			anim.PlayFetchIndex(RenderSprites.NormalizeSequence(anim, init.GetDamageState(), Sequence), () => 0);
+
+			yield return new SpriteActorPreview(anim, () => WVec.Zero, () => 0, p, rs.Scale);
+		}
+	}
+
+	public class WithSiloSpriteBody : WithSpriteBody, INotifyOwnerChanged
```

Comment:
Sorry to keep on dragging this out, but while giving this one final look over before merging I noticed the naming inconsistency between this and `WithResourceOverlay`.  Can we please rename this trait to `WithResourcesSpriteBody` and `WithResourceOverylay` to `WithResourcesOverlay`? This has an additional bonus of helping towards #6336.